src/Albany_Application.cpp

```diff
@@ -1358,6 +1358,15 @@ Application::computeGlobalResidualImpl(
            << ", computeGlobalResid workset.x = \n ";
       describe(workset.x.getConst(), *out, Teuchos::VERB_EXTREME);
 #endif
+
+      for (int ws = 0; ws < numWorksets; ws++) {
+        const std::string evalName = PHAL::evalName<EvalT>("FM", wsPhysIndex[ws]);
+        loadWorksetBucketInfo<EvalT>(workset, ws, evalName);
```

Comment:
I think this will break the memoization for parameter dependencies because `evalName` needs to be a unique string (i.e. different from the `evalName` used in `evaluateFields`). If you want to use memoization for `preEvaluate` and `evaluateFields` you'll have to create unique names and add it to PHAL::Setup (e.g. `"preEvalFM"` and `"evalFM"`). If you don't really need memoization in `preEvaluate`, just pass it a dummy string so it doesn't affect `evaluateFields` (e.g. `evalName = "preEvaluate"`).

Reply:
OK, thanks, I'll pass a dummy string for now.